### PR TITLE
DT-1195 and DT-1196 Add skeleton resource for POM delete and RO switc…

### PR DESCRIPTION
**Pull request by andymarke**
…hing

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource.java [hunk 1]
```diff
@@ -310,6 +312,30 @@ public CommunityOrPrisonOffenderManager allocatePrisonOffenderManagerByNomsNumbe
                 .orElseThrow(() -> new NotFoundException(String.format("Offender with noms number %s not found", nomsNumber)));
     }
 
+    @RequestMapping(value = "/offenders/nomsNumber/{nomsNumber}/prisonOffenderManager", method = RequestMethod.DELETE, consumes = "application/json")
+    @ApiResponses(value = {
+            @ApiResponse(code = 401, message = "Request is missing Authorization header (no JWT)"),
```
Comment (mikehalmamoj) [inline src/main/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource.java:317]: Needs a 403 too?

File: src/main/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource.java [hunk 2]
```diff
@@ -310,6 +312,30 @@ public CommunityOrPrisonOffenderManager allocatePrisonOffenderManagerByNomsNumbe
...
+            @ApiResponse(code = 404, message = "The offender is not found")
+    })
+    @ApiOperation(value = "Deallocates the prison offender manager for an offender in custody. The POM is set back to its unallocated state", notes = "Requires role ROLE_COMMUNITY_CUSTODY_UPDATE")
+    @PreAuthorize("hasRole('ROLE_COMMUNITY_CUSTODY_UPDATE')")
+    public void deallocatePrisonOffenderManagerByNomsNumber(final @PathVariable String nomsNumber) {
+    }
+
+    @RequestMapping(value = "/offenders/nomsNumber/{nomsNumber}/responsibleOfficer/switch", method = RequestMethod.PUT, consumes = "application/json")
+    @ApiResponses(value = {
+            @ApiResponse(code = 400, message = "Either set true for the prisoner offender manager or the community offender manager"),
```
Comment (mikehalmamoj) [inline src/main/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource.java:327]: Also needs 403?

File: src/main/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource.java (hunk 1 above)
Comment (andymarke) [reply src/main/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource.java:317]: agreed, I will add

File: src/main/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource.java (hunk 2 above)
Comment (andymarke) [reply src/main/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource.java:327]: agreed, I will add

File: src/test/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource_SwitchResponsibleOfficerAPITest.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+import uk.gov.justice.digital.delius.service.UserService;
+
+import static io.restassured.config.EncoderConfig.encoderConfig;
+import static io.restassured.module.mockmvc.RestAssuredMockMvc.given;
+import static io.restassured.module.mockmvc.config.RestAssuredMockMvcConfig.newConfig;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+@ExtendWith(MockitoExtension.class)
+public class OffendersResource_SwitchResponsibleOfficerAPITest {
```
Comment (mikehalmamoj) [inline src/test/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource_SwitchResponsibleOfficerAPITest.java:31]: Is it worth adding a 401 and 403 test now so it's not forgotten later?

Comment (mikehalmamoj) [reply src/test/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource_SwitchResponsibleOfficerAPITest.java:31]: Also maybe something for the DELETEs?

Comment (andymarke) [reply src/test/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource_SwitchResponsibleOfficerAPITest.java:31]: I was going to add those when I do an integration test, which will be first task

**Pull request merged**